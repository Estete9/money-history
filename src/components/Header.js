import { NavLink } from 'react-router-dom';
import React from 'react';

function Header() {
  return (
    <header>
      <NavLink to="/">
        <img src="/" alt="back button" />
        <h3>2005</h3>
      </NavLink>
      <h3>Currency</h3>
    </header>
  );
}

export default Header;
