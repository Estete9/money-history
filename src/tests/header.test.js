import React from 'react';
// import { render, fireEvent, cleanup } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Header from '../components/Header';

const mockStore = configureStore([]);

describe('Integration tests for Header component', () => {
  test('it renders correctly in CurrencyConversion component', () => {
    const store = mockStore({
      header: {
        isSearchBarOpen: false,
      },
    });

    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/currency/AUD']}>
          <Header />
        </MemoryRouter>
      </Provider>,
    );

    const headerTree = component.toJSON();
    expect(headerTree).toMatchSnapshot();
  });

  test('it renders correctly in CurrenciesList component', () => {
    const store = mockStore({
      header: {
        isSearchBarOpen: false,
      },
    });

    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      </Provider>,
    );

    const headerTree = component.toJSON();
    expect(headerTree).toMatchSnapshot();
  });
});
