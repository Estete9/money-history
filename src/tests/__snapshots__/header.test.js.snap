// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration tests for Header component it renders correctly in CurrenciesList component 1`] = `
<header
  className="header"
>
  <h3
    className="headerTitle"
  >
    Top 8 Currencies
  </h3>
  <button
    className="searchBtn"
    onClick={[Function]}
    type="button"
  >
    <img
      alt="search icon"
      className="searchIconImg"
      src="svg-mock"
    />
  </button>
</header>
`;

exports[`Integration tests for Header component it renders correctly in CurrencyHistory component 1`] = `
<header
  className="header"
>
  <a
    className="backSection"
    href="/"
    onClick={[Function]}
  >
    <img
      alt="back button"
      className="backBtn"
      src="svg-mock"
    />
    <h3
      className="backBtnText"
    >
      BACK
    </h3>
  </a>
  <h3
    className="headerTitle"
  >
    Top 8 Currencies
  </h3>
</header>
`;
